src/components/ModalWithForm.js

```diff
@@ -29,7 +29,9 @@ export class ModalWithForm extends Modal {
 
   changeEventListener(callback) {
     this._submitEventHandler = callback;
-    this.setEventListeners();
+    // this.eliminateEventListener();
```

Comment:
`eliminateEventListener` is not needed

* I can't delete cards. Nothing happens with I confirm deleting 


I wrote you last time what should be here


```js

  setEventListeners() {
    super.setEventListeners();
    this._form.addEventListener("submit", (evt) => {
      this._submitEventHandler(evt)  // you need to call the handler like this to be able to **update** it in `changeEventListener`
    });
  }

  changeEventListener(callback) {
    this._submitEventHandler = callback;
  }

```